tests/kernel/semaphore/semaphore/src/main.c

```diff
@@ -20,16 +21,16 @@
 
 struct timeout_info {
 	u32_t timeout;
-	struct k_sem *sema;
+	struct sys_sem *sema;
```

Comment:
I don't think we should change the test to use sys_sem instead of k_sem.
On systems that support both, we need to test both.
Perhaps we could change this test to do something with function pointers so that we can test both without copying tons of code.